lib/OpenQA/SignalBlocker.pm

```diff
@@ -0,0 +1,35 @@
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+package OpenQA::SignalBlocker;
+use Mojo::Base -base, -signatures;
+use Scalar::Util qw(weaken);
+
+my @SIGNALS = qw(TERM INT);
+
+# delays execution of signal handlers while an instance exists
+sub new ($class, @attributes) {
+    my $self = $class->SUPER::new(@attributes);
```

Comment:
Also not sure about the convension and for sure not a big deal but `cls` is one is used in other languages. feel free to close the comment...